### ISSUE TITLE
Creating mobile workers is broken

### ISSUE THREAD

**Issue by bsharif** [WDDCP/commcare-wddcp#10]
Trying to create mobile workers is broken currently. 

This is accessed using: Users > Mobile Workers 

Page displays this error when first loading: 

>There was an issue contacting the server. Please check your internet connection. If this issue continues, please file a bug report.

![fireshot capture 3 - mobile workers _ users __ - commcare h_ - http___ec2-35-177-33-162 eu-west-2](https://user-images.githubusercontent.com/7132252/27769137-2eddd89e-5f1b-11e7-8616-c71b9a91be34.png)

[Dev group thread](https://groups.google.com/d/msg/commcare-developers/cjJ3RXsx7Uk/RF0DViJeAwAJ) suggests this may be due to Elasticsearch indexes not being initialized (although other user managed to fix by purging/resintalling Elasticsearch/indexes?)

[Another thread](https://groups.google.com/d/msg/commcare-developers/Xx0VAySFlYg/16HLArvzBAAJ) leads to this attached Word doc
[COMMCARE HQ  Setup instructions.docx](https://github.com/WDDCP/commcare-wddcp/files/1117740/COMMCARE.HQ.Setup.instructions.docx).
Suggests this may be the process for fixing mobile worker issues:

> Related to mobile worker list issues
> For the deletion of elasticsearch index
> `curl -XDELETE localhost:9200/.mar*`
> For recreation of elasticsearch index
>  `./manage.py ptop_preindex`

**Comment by bsharif** [comment]
Update on this.  It appears that **creating** mobile workers may not be broken, perhaps more retrieving the list from the server? 

In attached image I created a new mobile worker called 'michael'. 
![michael_create](https://user-images.githubusercontent.com/7132252/27769500-1723526c-5f23-11e7-8112-72da29b921fe.png)
The link for the URL of the username (michael) is: 
[http://ec2-35-177-33-162.eu-west-2.compute.amazonaws.com:8000/a/demo/settings/users/commcare/account/91652e11c6c9425ba2d73ea37c0dc9c1/](http://ec2-35-177-33-162.eu-west-2.compute.amazonaws.com:8000/a/demo/settings/users/commcare/account/91652e11c6c9425ba2d73ea37c0dc9c1/)
This leads to this error page:
![fireshot capture 8 - eserror at _a_demo_settings_users_comm_ - http___ec2-35-177-33-162 eu-west-2](https://user-images.githubusercontent.com/7132252/27769528-b4cae610-5f23-11e7-8bfc-4cd03ba06244.png)

And on going back/page refresh, the mobile worker I created (michael) is not displayed/retrieved correctly. 

However if I try to **recreate** the 'michael' mobile worker then I'm not allowed as the record must be existing somewhere...
![fireshot capture 9 - mobile workers _ users __ - commcare h_ - http___ec2-35-177-33-162 eu-west-2](https://user-images.githubusercontent.com/7132252/27769539-f904dcd2-5f23-11e7-83a5-5e32328f3530.png)

Hmm....

**Comment by bsharif** [comment]
Again, some more investigation. I'm confident that creation of mobile workers is working. 
If you check the Django Admin:
> Admin > Django Admin

then go:
> Authentication and Authorization › Users

then it lists the mobile workers I created. 

Then using the details you can login to CloudCare here:
http://ec2-35-177-33-162.eu-west-2.compute.amazonaws.com:8000/a/demo/login/

So it appears that the issue is less creation and more retrieval of mobile workers.

**Comment by DeckOfPandas** [comment]
I agree that it's an elasticsearch problem.

Stack trace for the server starts with:
`elasticsearch_1  | [2017-07-10 07:52:50,038][DEBUG][action.search.type       ] [Advisor] [hqusers_2017-05-22][1], node[EgLUTjpiS86x3Ql5wtYOgw], [P], s[STARTED]: Failed to execute [org.elasticsearch.action.search.SearchRequest@36efe4cf] lastShard [true]
elasticsearch_1  | org.elasticsearch.search.SearchParseException: [hqusers_2017-05-22][1]: from[-1],size[-1]: Parse Failure [Failed to parse source [{"sort": [{"username.exact": {"order": "asc"}}], "query": {"filtered": {"filter": {"and": [{"or": [{"term": {"domain.exact": "demo"}}, {"term": {"domain_memberships.domain.exact": "demo"}}]}, {"term": {"doc_type": "CommCareUser"}}, {"term": {"is_active": true}}, {"term": {"base_doc": "couchuser"}}]}, "query": {"match_all": {}}}}, "from": 0, "size": 0}]]`

And ends with:
`[action.search.type       ] [Advisor] All shards failed for phase: [query]`



Dimagi have helpfully provided scripts for managing some of the services, including elasticsearch. Here is the start of what I am looking into:

`root@44f1f0ffe890:/mnt/commcare-hq# ./scripts/elasticsearch-administer.py commands cluster_status`

`CLUSTER HEALTH
Traceback (most recent call last):
  File "./scripts/elasticsearch-administer.py", line 159, in <module>
    main()
  File "./scripts/elasticsearch-administer.py", line 155, in main
    commands[args.command](es)
  File "./scripts/elasticsearch-administer.py", line 47, in cluster_status
    pprint(cluster.health())
  File "/vendor/lib/python2.7/site-packages/elasticsearch/client/utils.py", line 69, in _wrapped
    return func(*args, params=params, **kwargs)
  File "/vendor/lib/python2.7/site-packages/elasticsearch/client/cluster.py", line 30, in health
    'health', index), params=params)
  File "/vendor/lib/python2.7/site-packages/elasticsearch/transport.py", line 307, in perform_request
    status, headers, data = connection.perform_request(method, url, params, body, ignore=ignore, timeout=timeout)
  File "/vendor/lib/python2.7/site-packages/elasticsearch/connection/http_urllib3.py", line 89, in perform_request
    raise ConnectionError('N/A', str(e), e)
elasticsearch.exceptions.ConnectionError: ConnectionError(<urllib3.connection.HTTPConnection object at 0x7f384e20fd90>: Failed to establish a new connection: [Errno -2] Name or service not known) caused by: NewConnectionError(<urllib3.connection.HTTPConnection object at 0x7f384e20fd90>: Failed to establish a new connection: [Errno -2] Name or service not known)`

And there's nothing at localhost:5555

In summary: elasticsearch doesn't seem to be running, but I don't know why not (yet)

**Comment by DeckOfPandas** [comment]
Screenshot of stack trace from #22 (duplicate)

![28238227-02b48c3c-6947-11e7-9483-49e4b6d4adab](https://user-images.githubusercontent.com/2399432/30822561-22d51cc6-a221-11e7-8140-6a3dd664a696.png)

**Comment by DeckOfPandas** [comment]
Oooh I bet it's this:
![screen shot 2017-09-25 at 20 53 32](https://user-images.githubusercontent.com/2399432/30827914-b4ab156c-a233-11e7-8540-59f9d54730a9.png)

**Comment by bsharif** [comment]
Also now fixed in @TheTincho 's build. Closed 👍
